cmake/projects/GTest/hunter.cmake

```diff
@@ -237,6 +249,7 @@ hunter_cmake_args(
     GTest
     CMAKE_ARGS
     HUNTER_INSTALL_LICENSE_FILES=${_gtest_license}
+    gtest_force_shared_crt=TRUE
```

Comment:
This seemed really odd to me, before I looked it up. Can you add a small comment explaining the use of this flag?

"This is unfortunate, but easily handled by setting the inaccurately named gtest_force_shared_crt option to TRUE (all this does is prevent GoogleTest from modifying options rather than force it to use shared libraries!)." from https://crascit.com/2015/07/25/cmake-gtest/

Reply:
I think I understand why. turning this `gtest_force_shared_crt` option to true basically forces googletest to do absolutely nothing regardless if you set `/MD` or `/MDd` or `/MT`  or `/MTd` in your CMAKE_CXX_FLAGS_<build type>.

check the code: https://github.com/google/googletest/blob/master/googletest/cmake/internal_utils.cmake#L33-L42

in other word, `gtest_force_shared_crt=ON` actually means _Leave my CRT flags*. It's not the first time i'm seeing this from goodle. I encountered the same exact issue here in Google's shaderc: https://github.com/google/shaderc/blob/master/CMakeLists.txt#L69-L83

Reply:
sorry, I just re-read @rbsheth and we're both saying the same thing lol